src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -3938,12 +3940,23 @@ public override DynamicMetaObject FallbackGetIndex(DynamicMetaObject target, Dyn
                         return result.WriteToDebugLog(this);
                     }
                 }
+
+                // If the type explicitly implements an indexer specified by an interface
+                // then the DefaultMemberAttribute will not carry over to the implementation.
+                // This check will catch those cases.
+                if (defaultMember == null)
+                {
+                    defaultMember = i.GetCustomAttribute<DefaultMemberAttribute>(inherit: true);
```

Comment:
A minor comment: Attributes cannot be inherited from interfaces, so it should be `inherit: false`.